components/availability/src/Availability.svelte

```diff
@@ -254,12 +255,32 @@
   const dispatchEvent = getEventDispatcher(get_current_component());
 
   //#region layout
+
   let main: HTMLElement;
   let tickContainer: HTMLElement;
+  let dayContainerWidth: number = 0;
   let clientHeight: number;
 
+  const MINIMUM_DAY_WIDTH = 100;
+
+  // Internally-settable reactive-to-external-props variable
+  let datesToShow: number;
+
+  $: optimalDatesToShow =
+    Math.floor(dayContainerWidth / MINIMUM_DAY_WIDTH) || 1;
+  $: datesToShow =
+    optimalDatesToShow < dates_to_show || show_as_week
+      ? optimalDatesToShow
+      : dates_to_show;
+
+  $: tooSmallForWeek = show_as_week && optimalDatesToShow < 7;
+
+  //#endregion layout
+
+  //#region data generation
```

Comment:
side-effect: I've broken up the `layout` region into layout and `data generation` for easier compartmentalizing / later modules.